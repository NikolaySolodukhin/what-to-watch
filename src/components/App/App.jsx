import React from 'react';
import {Switch, Route} from 'react-router-dom';

import WelcomeScreen from './../WelcomeScreen/WelcomeScreen.jsx';
import {PageSignIn} from './../PageSignIn/PageSignIn.jsx';
import PageMyList from './../PageMyList/PageMyList.jsx';
import WithAuthRoute from './../hocs/WithAuthRoute/WithAuthRoute.jsx';
import PageFilm from './../PageFilm/PageFilm.jsx';
import WithOutAuthRoute from './../hocs/WithOutAuthRoute/WithOutAuthRoute.jsx';
import PageFilmReview from './../PageFilmReview/PageFilmReview.jsx';

function App() {
  return (
    <Switch>
      <Route exact path="/" component={WelcomeScreen}/>
      <Route exact path="/film/:id/review" component={PageFilmReview}/>
      <Route path="/film/:id" component={PageFilm}/>
      <WithAuthRoute path="/my-list" component={PageMyList}/>
      <WithOutAuthRoute exact path="/login" component={PageSignIn}/>
    </Switch>
  );
}

export {App};
