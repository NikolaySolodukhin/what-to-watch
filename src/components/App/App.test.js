import React from 'react';
import renderer from 'react-test-renderer';

import App from './App.jsx';

const FILMS_LIST = [{
  id: 1,
  title: `Fantastic Beasts`,
  image: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`
}, {
  id: 2,
  title: `Bohemian Rhapsody`,
  image: `img/bohemian-rhapsody.jpg`
}, {
  id: 3,
  title: `Macbeth`,
  image: `img/macbeth.jpg`
}, {
  id: 4,
  title: `Aviator`,
  image: `img/aviator.jpg`
}, {
  id: 5,
  title: `We need to talk about Kevin`,
  image: `img/we-need-to-talk-about-kevin.jpg`
}];

it(`App is rendered correctly`, () => {
  const tree = renderer.create(<App
    films={FILMS_LIST}
  />).toJSON();

  expect(tree).toMatchSnapshot();
});
