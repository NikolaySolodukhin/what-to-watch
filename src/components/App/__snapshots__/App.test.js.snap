// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App/> App renders correctly 1`] = `
<Switch>
  <Route
    component={
      Object {
        "$$typeof": Symbol(react.memo),
        "WrappedComponent": [Function],
        "compare": null,
        "displayName": "Connect(WelcomeScreen)",
        "type": [Function],
      }
    }
    exact={true}
    path="/"
  />
  <Route
    component={
      Object {
        "$$typeof": Symbol(react.memo),
        "WrappedComponent": [Function],
        "compare": null,
        "displayName": "Connect(PageFilmReview)",
        "type": [Function],
      }
    }
    exact={true}
    path="/film/:id/review"
  />
  <Route
    component={
      Object {
        "$$typeof": Symbol(react.memo),
        "WrappedComponent": [Function],
        "compare": null,
        "displayName": "Connect(PageFilm)",
        "type": [Function],
      }
    }
    path="/film/:id"
  />
  <ConnectFunction
    component={
      Object {
        "$$typeof": Symbol(react.memo),
        "WrappedComponent": [Function],
        "compare": null,
        "displayName": "Connect(PageMyList)",
        "type": [Function],
      }
    }
    path="/my-list"
  />
  <ConnectFunction
    component={[Function]}
    exact={true}
    path="/login"
  />
</Switch>
`;
